server/trace/mock_trace_test.go

```diff
@@ -203,6 +203,33 @@ func TestMockTraceServer_BatchWriteSpans_InvalidTimestamp(t *testing.T) {
 	}
 }
 
+func TestMockTraceServer_BatchWriteSpans_DuplicateName(t *testing.T) {
+	setup()
+	defer tearDown()
+
+	const duplicateSpanName = "test-span-1"
+	spans := []*cloudtrace.Span{
+		generateSpan(duplicateSpanName),
+		generateSpan(duplicateSpanName),
+	}
+	in := &cloudtrace.BatchWriteSpansRequest{Name: "test-project", Spans: spans}
+	want := validation.ErrDuplicateSpanName
+
+	responseSpan, err := client.BatchWriteSpans(ctx, in)
+	if err == nil {
+		t.Errorf("BatchWriteSpans(%q) == %q, expected error %q",
+			in, responseSpan, want.Err())
```

Comment:
nit: prefer early return and omit the else

Reply:
Done, fixed this for other tests too